agreement/views/agreement.xml

```diff
@@ -31,19 +31,20 @@
                     <group name="main">
                         <group name="left">
                             <field name="code" />
+                            <field
+                                name="is_template"
+                                groups="agreement.group_use_agreement_template"
+                            />
                             <field
                                 name="partner_id"
                                 attrs="{'required': [('is_template', '=', False)]}"
+                                groups="agreement.group_use_agreement_template"
```

Comment:
He added this group because the attr need the value of _is_template_ but it is protected by a group. I prefer to remove the group on partner_id and add before the field is_template with invisible="1". In this way, the result of the attr can be evaluated by odoo.
```suggestion
                                <field name="is_template" invisible="1 />
                                <field
                                name="partner_id"
                                attrs="{'required': [('is_template', '=', False)]}"
                                />
```